src/cloud-api-adaptor/pkg/userdata/provision.go

```diff
@@ -149,60 +164,82 @@ func parseUserData(userData []byte) (*CloudConfig, error) {
 	return &cc, nil
 }
 
-func findConfigEntry(path string, cc *CloudConfig) []byte {
-	for _, wf := range cc.WriteFiles {
-		if wf.Path != path {
-			continue
-		}
-		return []byte(wf.Content)
+func writeFile(path string, bytes []byte) error {
+	// Ensure the parent directory exists
+	err := os.MkdirAll(filepath.Dir(path), 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create directory: %w", err)
+	}
+
+	err = os.WriteFile(path, bytes, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write file %s: %w", path, err)
 	}
+	logger.Printf("Wrote %s\n", path)
 	return nil
 }
 
-type entry struct {
-	path     string
-	optional bool
+func processCloudConfig(cfg *Config, cc *CloudConfig) error {
+	for _, wf := range cc.WriteFiles {
```

Comment:
I would not write arbitrary files in cloud-config, but just the ones that we expect, so a specific allow-list with path `/run/peerpod/daemon.json`, `/run/peerpod/auth.json`, ...

Reply:
Yes, added allowlist for items can be in initdata and write_files.